components/channel_header/channel_header.js

```diff
@@ -286,6 +299,72 @@ class ChannelHeader extends React.PureComponent {
 
     handleFormattedTextClick = (e) => Utils.handleFormattedTextClick(e, this.props.currentRelativeTeamUrl);
 
+    showCustomStatusTextTooltip = () => {
+        const element = this.customStatusTextRef;
+        if (element && element.offsetWidth < element.scrollWidth) {
+            this.setState({showCustomStatusTooltip: true});
+        } else {
+            this.setState({showCustomStatusTooltip: false});
+        }
+    }
+
+    renderCustomStatus = () => {
+        const customStatus = this.props.customStatus;
+        const isStatusSet = customStatus && (customStatus.text || customStatus.emoji);
+        if (!(this.props.isCustomStatusEnabled && isStatusSet)) {
+            return null;
+        }
+        let customStatusTextComponent = (
+            <span
+                className='overflow--ellipsis text-nowrap'
+                ref={(element) => {
+                    this.customStatusTextRef = element;
+                    this.showCustomStatusTextTooltip();
+                }}
+            >
+                <Markdown
+                    message={customStatus.text}
```

Comment:
@hmhealey I see your point. Allowing markdown here allows custom statuses which are images/Gifs, which was not our original intention.

We used markdown here to allow emojis to be rendered in the custom status messages.

![Screenshot from 2021-02-15 19-35-12](https://user-images.githubusercontent.com/35728906/107956412-22eb0480-6fc5-11eb-9c8a-ddf096479dcb.png)

This requirement was mentioned somewhere in the spec. If we remove it here, we won't be able to allow support for that feature.

@esethna @matthewbirtch thoughts?

Reply:
Since the status already has an emoji as part of it, I'd lean towards just using that. We could always go back and add emoji support later on if we decide we want it too since it's a bit easier to add than remove.

That said, I'll defer to Eric and Matt here, and I'll make sure they see this

Reply:
@chetanyakan I'd propose we don't support markdown for now in the status text. This opens up too many corner cases to validate now. We may include this as a future update